bodhi/server/consumers/masher.py

```diff
@@ -917,53 +1077,6 @@ def sort_by_days_in_testing(self, updates):
         updates.sort(key=lambda update: update.days_in_testing, reverse=True)
         return updates
 
-    @checkpoint
-    def compose_atomic_trees(self):
-        """Compose Atomic OSTrees for each tag that we mashed."""
-        composer = AtomicComposer()
-        mashed_repos = dict([('-'.join(os.path.basename(repo).split('-')[:-1]), repo)
-                             for repo in self.state['completed_repos']])
-        for tag, mash_path in mashed_repos.items():
-            if tag not in atomic_config['releases']:
-                log.warn('Cannot find atomic configuration for %r', tag)
-                continue
-
-            # Update the repo URLs to point to our local mashes
-            release = copy.deepcopy(atomic_config['releases'][tag])
-            mash_path = 'file://' + os.path.join(mash_path, tag, release['arch'])
-
-            if 'updates-testing' in tag:
-                release['repos']['updates-testing'] = mash_path
-                updates_tag = tag.replace('-testing', '')
-                if updates_tag in mashed_repos:
-                    release['repos']['updates'] = 'file://' + os.path.join(
-                        mashed_repos[updates_tag], updates_tag,
-                        release['arch'])
-                log.debug('Using the updates repo from %s',
-                          release['repos']['updates'])
-            else:
-                release['repos']['updates'] = mash_path
-
-            # Compose the tree, and raise an exception upon failure
-            notifications.publish(topic="ostree.compose.start",
-                                  msg=dict(tag=tag),
-                                  force=True)
-            result = composer.compose(release)
-            if result['result'] != 'success':
-                self.log.error(result)
-                notifications.publish(topic="ostree.compose.fail",
-                                      msg=dict(tag=tag),
-                                      force=True)
-                raise Exception('%s atomic compose failed' % tag)
-            else:
-                self.log.info('%s atomic tree compose successful', tag)
-                notifications.publish(
-                    topic="ostree.compose.finish",
-                    msg=dict(tag=tag,
-                             ref=result.get('ref'),
-                             commitid=result.get('commitid')),
-                    force=True)
```

Comment:
I can't wait to kill fedmsg-atomic-composer (which, humorously, isn't triggered via fedmsgs).

Reply:
Well, fedmsg-atomic-composer does have a fedmsg consumer and can be ran through purely fedmsg.... :)